sdk/appconfiguration/app-configuration/test/public/index.spec.ts

```diff
@@ -1140,4 +1155,308 @@ describe("AppConfigurationClient", () => {
       });
     });
   });
+
+  describe("FeatureFlag configuration setting", () => {
+    const clientFilters: (
+      | Record<string, unknown>
+      | FeatureFlagTargetingClientFilter
+      | FeatureFlagTimeWindowClientFilter
+      | FeatureFlagPercentageClientFilter
+    )[] = [
+      {
+        name: "Microsoft.TimeWindow",
+        parameters: {
+          start: "Wed, 01 May 2019 13:59:59 GMT",
+          end: "Mon, 01 July 2019 00:00:00 GMT"
+          // TODO: dates should accept `string | Date`
+          // Returning
+          //      - SDK would return Date for all the config-settings if parsable
+          //      - SDK would return strings for non parsable strings
+          // Passing as an argument
+          //      - If passed in a string, we'll pass it as is.
+          //      - If passed Date, we'll make it a string.
+        }
+      },
+      { name: "FilterX" },
+      {
+        name: "Microsoft.Targeting",
+        parameters: {
+          audience: {
+            groups: [
+              { name: "group-1", rolloutPercentage: 25 },
+              { name: "group-2", rolloutPercentage: 45 }
+            ],
+            users: ["userA", "userB"],
+            defaultRolloutPercentage: 40
+          }
+        }
+      },
+      { name: "Microsoft.Percentage", parameters: { value: 25 } }
+    ];
+
+    let baseSetting: FeatureFlag;
+    let addResponse: AddConfigurationSettingResponse;
+
+    beforeEach(async () => {
+      baseSetting = {
+        conditions: {
+          clientFilters
+        },
+        enabled: false,
+        isReadOnly: false,
+        key: `${featureFlagPrefix + recorder.getUniqueName("name-1")}`,
```

Comment:
So, this is where JS deviates from the rest of the languages. But, this is the best option I guess. @richardpark-msft 

cc @mssfang @pakrym @seankane-msft

Reply:
What is the deviation?

Reply:
Other languages take the featureId and add the prefix to form the key.
We ask users to provide the key.

Reply:
Given our JS nature, we could say "well, you're specifying feature flag things so we'll add-prefix-if-not-exist". What do you think of that for post-preview?

Reply:
Sounds great, will add in a future PR.

Reply:
Logged an issue